JeMPI_Apps/JeMPI_UI/tests/hooks/useAuth.test.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * @jest-environment jsdom
+ */
+
+process.env.REACT_APP_ENABLE_SSO = 'true'
+
+// js-dom does not implement fetch
+import { enableFetchMocks } from 'jest-fetch-mock'
+enableFetchMocks();
+
+import React, { ReactNode } from "react"
+import {useAuth } from "../../src/hooks/useAuth"
+import {  act, render, screen, waitFor } from '@testing-library/react'
+
+import { QueryClient } from "@tanstack/react-query"
+import { SnackbarProvider } from "notistack"
+import ApiClient from "services/ApiClient"
+import lodash from "lodash"
```

Comment:
I don't think this will necessarily add 70kb because of tree shaking https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking i.e only the used function will be bundled. What is true though is that the way I have imported it would not allow for tree shaking (and indeed it would result in the 70kb being bundled). Should be more specific.

Regarding writing our own `omit` function, this possible, but from experince having a robust utillies libray that focus on items like this is usually better. They focus on optimizing those functions, and ensuring they are working as should. Unless there is a signficate benefit for not using a (robust) utilities libray, I see no reason not to.

Reply:
Hi @walisc please take a look at this repo 
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore

Reply:
Hi @issambaccouch . Thanks for this, and interesting repo. I still maintain that we need a library like `lodash`. As  indicated in their readme `...and you know your target browser **clearly**, then you may not need Lodash/Underscore.` . We actually don't know what we are targeting (and dont think we can ever fully know beforehand). Is it a traditional browser or not? (i.e some custom browser for security reasons). Are they using the latest version or not? (with natonal-level software we can't take that for granted). Are we running this within the openHIM platform (as a webview) or not etc

Having to rewrite these functions. ensure that they work correctly (i.e we are not introducing bugs), on various browsers, and maintaining them as well is a lot of unnecessary work I feel. If they are people dedicated to that (and have a lot of credibility) why not delegate to them/use them.

Reply:
Hi @issambaccouch I have tried to use the omit function recommended at the link you mentioned above (https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore), and this is not working. The reason being that the authContext object has some non-enumerable properties (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties), which means a the spread operator wont pick these properties up (or a for ...in ). Is there is any other (simple) native way of doing this?

Reply:
Have removed lodash for now. p.s re:- non-enumerable properties (if you saw that message). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties This is one place that a native omit function could fail. The authContext does not have any non-enumerable properties (as previously indicated) hence a native function can work for now.